src/utils/DateUtils.ts

```diff
@@ -29,111 +130,174 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
   predefinedFormat: string = undefined;
 }
 
+/**
+ * This class modifies the received Date so that it is rendered the way the options specify.
+ */
 export class DateUtils {
-  static convertFromJsonDateIfNeeded(date: string): Date;
-  static convertFromJsonDateIfNeeded(date: number): Date;
-  static convertFromJsonDateIfNeeded(date: Date): Date;
+
   static convertFromJsonDateIfNeeded(date: any): Date {
+    return DateUtils.convertToStandardDate(date);
+  }
+
+  static convertToStandardDate(date: string): Date;
+  static convertToStandardDate(date: number): Date;
+  static convertToStandardDate(date: Date): Date;
+  /**
+   * Receives a date of any type and converts it to a standard Date object.
+   * @param date can be a string, a number or a Date object. It can also be any other type, but the result will be
+   * 'undefined' if it is not recognized as a date.
+   * @returns {any} the Date, or undefined it it wasn't recognized as a valid date.
+   */
+  static convertToStandardDate(date: any): Date {
     if (_.isDate(date)) {
-      return date;
+      return moment(date).toDate();
     } else if (date !== null && !isNaN(Number(date))) {
-      return new Date(Number(date));
+      return moment(Number(date)).toDate();
     } else if (_.isString(date)) {
-      return new Date(<string>date.replace('@', ' '));
+      /*
+      If the input is a string recognized as a Date, parse it using momentJS.
+      This array specifies the possible string formats the moment will have to parse. It will be either the standard
+      ISO_8601 or the format our index uses to store dates ('YYYY/MM/DD@HH:mm:ssZ').
+      */
```

Comment:
I think the first sentence of this comment is not necessary, it's explicit when reading the code. Also I think you could completely remove this comment by assigning the array to a meaningful variable name. For comments like this we use `//` because `/*` is used to generate JSDoc.